src/core/expr/head_reduce_unary.cc

```diff
@@ -352,6 +352,87 @@ See Also
 
 - :func:`count()` -- function to calculate a number of non-missing values.
 
+Examples
+--------
+
+::
+
+    from datatable import dt, f, by
+
+    df = dt.Frame({'A': [1, 1, 2, 1, 2],
+                   'B': [None, 2, 3,4, 5],
+                   'C': [1, 2, 1, 1, 2]})
+
+    df
+
+.. dtframe::
+    :names: A,B,C
+    :types: int8, int8, int8
+    :shape: 5, 2
+
+    0,1,NA,1
+    1,1,2,2
+    2,2,3,1
+    3,1,4,1
+    4,2,5,2
+
+Get the sum of column A::
+
+    df[:, dt.sum(f.A)]
+
+.. dtframe::
+    :names: A
+    :types: int8
+    :shape: 1, 1
+
+    0,7
+
+Get the sum of multiple columns::
+
+  df[:, [dt.sum(f.A), dt.sum(f.B)]]
+
+.. dtframe::
+    :names: A,B
+    :types: int8,int8
+    :shape: 1, 2
+
+    0,7,14
+
+
+Same as above, but more convenient::
+
+  df[:, dt.sum(f[:2])]
+
+.. dtframe::
+    :names: A,B
+    :types: int8,int8
+    :shape: 1, 2
+
+    0,7,14
+
+
+In the presence of :func:`by()`, it returns the sum of the specified columns per group::
+
+  df[:, [dt.sum(f.A), dt.sum(f.B)], by(f.C)]
+
+.. dtframe::
+    :names: C,A,B
+    :types: int8,int8,int8
+    :shape: 2, 3
+
+    0,1,4,7
+    1,2,3,7
+
+
+
+:func:`sum()` acts like python's built-in ``sum`` function if the value is not a :ref:`f-expressions`::
```

Comment:
We will be changing this by the 1.0 release, so that all reducer functions have the same API.
I guess it's ok to leave this example as-is (given that this is how the function behaves right now), on the other hand we wouldn't want users to start relying on this behavior if it's going to be removed in the next release.